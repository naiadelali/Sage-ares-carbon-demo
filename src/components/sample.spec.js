import React from 'react';
import { shallow } from 'enzyme';
import Button from 'carbon-react/lib/components/button';
import Sample from './sample';

describe('Sample Component', () => {
  const renderShallow = () => (
    shallow(<Sample />)
  );

  it('renders a button', () => {
    const wrapper = renderShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('triggers an alert on click', () => {
    const wrapper = renderShallow();
    spyOn(global, 'alert');
    wrapper.find(Button).simulate('click');
    expect(global.alert).toHaveBeenCalledWith('You clicked the button!');
  });
});
