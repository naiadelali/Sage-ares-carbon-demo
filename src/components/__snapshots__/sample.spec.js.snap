// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Component renders a button 1`] = `
<Button
  as="secondary"
  disabled={false}
  onClick={[Function]}
  size="medium"
  subtext=""
  theme="blue"
>
  An Example Button!
</Button>
`;
